datasets/greek_legal_code/greek_legal_code.py

```diff
@@ -0,0 +1,149 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""\
+Greek_Legal_Code contains 47k classified legal resources from Greek Legislation. Its origin is “Permanent Greek Legislation Code - Raptarchis”,
+a collection of Greek legislative documents classified into multi-level (from broader to more specialized) categories.
+"""
+
+
+import json
+import os
+
+import datasets
+
+
+_CITATION = """\
+@inproceedings{papaloukas-etal-2021-glc,
+    title = "Multi-granular Legal Topic Classification on Greek Legislation",
+    author = "Papaloukas, Christos and Chalkidis, Ilias and Athinaios, Konstantinos and Pantazi, Despina-Athanasia and Koubarakis, Manolis",
+    booktitle = "Proceedings of the 3rd Natural Legal Language Processing (NLLP) Workshop",
+    year = "2021",
+    address = "Punta Cana, Dominican Republic",
+    publisher = "",
+    url = "",
+    doi = "",
+    pages = ""
+}
+"""
+
+_DESCRIPTION = """\
+Greek_Legal_Code contains 47k classified legal resources from Greek Legislation. Its origin is “Permanent Greek Legislation Code - Raptarchis”,
+a collection of Greek legislative documents classified into multi-level (from broader to more specialized) categories.
+"""
+
+_HOMEPAGE = "https://doi.org/10.5281/zenodo.5528002"
+
+_LICENSE = "CC BY-SA (Creative Commons / Attribution-ShareAlike)"
+
+URL = "https://zenodo.org/record/5528002/files/greek_legislation.tar.xz"
+
+
+class GreekLegalCodeConfig(datasets.BuilderConfig):
+    """BuilderConfig for Greek_Legal_Code."""
+
+    def __init__(self, label_type="volume", **kwargs):
+        """BuilderConfig for Greek_Legal_Code.
+
+        Args:
+        label_type: One of following: [volume, chapter, subject]
+          **kwargs: keyword arguments forwarded to super.
+        """
+        super(GreekLegalCodeConfig, self).__init__(**kwargs)
+        self.label_type = label_type
+
+
+class GreekLegalCode(datasets.GeneratorBasedBuilder):
+    """\
+    Greek_Legal_Code contains 47k classified legal resources from Greek Legislation. Its origin is “Permanent Greek Legislation Code - Raptarchis”,
+    a collection of Greek legislative documents classified into multi-level (from broader to more specialized) categories.
+    """
+
+    VERSION = datasets.Version("1.0.0")
+
+    BUILDER_CONFIGS = [
+        GreekLegalCodeConfig(
+            name=label_type,
+            label_type=label_type,
+            version=datasets.Version("1.0.0", ""),
+            description=f"Plain text import of Greek_Legal_Code for the {label_type} level",
+        )
+        for label_type in ["volume", "chapter", "subject"]
+    ]
+
+    DEFAULT_CONFIG_NAME = "volume"
+
+    def _info(self):
+        features = datasets.Features(
+            {
+                "text": datasets.Value("string"),
+                "label": datasets.Value("string"),
```

Comment:
it would be nice to have the list of the possible labels here (even though they can be numerous in the "subject" case):
```suggestion
                "label": datasets.ClassLabel(names=[...]),
```